src/mockServiceWorker.js

```diff
@@ -200,7 +200,8 @@ If you wish to mock an error response, please refer to this guide: https://mswjs
             ok: clonedResponse.ok,
             status: clonedResponse.status,
             statusText: clonedResponse.statusText,
-            body: await clonedResponse.text(),
+            body:
+              clonedResponse.body === null ? null : await clonedResponse.text(),
```

Comment:
And we don't need the same logic on the line 124 because `body` in the mocked response is `null` by default?

https://github.com/mswjs/msw/blob/9c894760fef38f84d8fd4841a837a5ab6e1c5ec7/src/response.ts#L37

Reply:
ah yes, I didn't see it. I think that we can add the same code there and add a new test

Reply:
Not sure about adding the same code there. Line 124 is controlled by the end developer. If they provide a body or an empty string as a body, for a 204 response they are making a mistake. It would be good for the underlying logic to tell them about it, instead of coercing their mistakes. 

In case of the `response` event (to which this entire `.then` clause is dedicated), the mistake is on our side, because the end developer doesn't directly construct the `Response` instance, instead they act as a consumer of an already valid response instance.

Reply:
> Not sure about adding the same code there. Line 124 is controlled by the end developer. If they provide a body or an empty string as a body, for a 204 response they are making a mistake. It would be good for the underlying logic to tell them about it, instead of coercing their mistakes.

Maeby we can create a real `Response` object somewhere, for example in `createResponseComposition`. In case of error we could inform the user of the issue.

Reply:
We can, but it would be only for that purpose alone, as we cannot send the `Response` instance to the worker, it needs to be converted into JSON beforehand. I think it's not the issue that the error happens in the worker, if that's your concern.

Reply:
yes yes I know it was only for a check. Surround `Response` with a try-catch. If an error is caught we could inform the user otherwise we will continue as is

Reply:
This is of course not regarding to this issue.

I mean something like

```diff
diff --git a/src/response.ts b/src/response.ts
index 5fe12a4..6590907 100644
--- a/src/response.ts
+++ b/src/response.ts
@@ -67,11 +67,18 @@ export function createResponseComposition<BodyType>(
       ...transformers,
     ].filter(Boolean)
 
-    const resolvedResponse =
-      resolvedTransformers.length > 0
-        ? compose(...resolvedTransformers)(initialResponse)
-        : initialResponse
+    const resolvedResponse = await (resolvedTransformers.length > 0
+      ? compose(...resolvedTransformers)(initialResponse)
+      : initialResponse)
 
+    try {
+      new Response(resolvedResponse.body, { ...resolvedResponse })
+    } catch (e) {
+      throw new Error(`
+There is an issue with the response in your handler. This is the error thrown:
+"{e}"
+In most cases your body has a conflict with status. Make sure that your are following the principles of fetch. See https://fetch.spec.whatwg.org/#responses`)
+    }
     return resolvedResponse
   }
 }
```

Reply:
However, the same (but less user-friendly) error would surface to the user anyway. I'm not sure this extra check is necessary. Everybody will pay for it, as the `Response` that's not used will be constructed for each `res()` invocation, but only those who make mistakes would get the benefit of this check. Moreover, this check would guard against a spec-incompliant usage more than an invalid usage of MSW. I'd say this check is not necessary as of now.

Reply:
for me is ok